packages/common/src/web3Proofs.ts

```diff
@@ -0,0 +1,596 @@
+    if (statements.length === 0) {
+        throw new Error('Empty statements are not allowed');
+    }
+    const checkedStatements = [];
+    for (const s of statements) {
+        verifyAtomicStatement(s, checkedStatements, schema);
+        checkedStatements.push(s);
+    }
+    return true;
+}
+
+function getWeb3IdCredentialQualifier(
+    validContractAddresses: ContractAddress[]
+): VerifiableCredentialQualifier {
+    return {
+        type: 'sci',
+        issuers: validContractAddresses,
+    };
+}
+
+function getAccountCredentilQualifier(
+    validIdentityProviders: number[]
+): IdentityQualifier {
+    return {
+        type: 'cred',
+        issuers: validIdentityProviders,
+    };
+}
+
+export class AtomicStatementBuilder implements InternalBuilder {
+    statements: AtomicStatementV2[];
+    schema: VerifiableCredentialSubject | undefined;
+
+    constructor(schema?: VerifiableCredentialSubject) {
+        this.statements = [];
+        this.schema = schema;
+    }
+
+    /**
+     * Outputs the built statement.
+     */
+    getStatement(): AtomicStatementV2[] {
+        return this.statements;
+    }
+
+    /**
+     * If checkConstraints is true, this checks whether the given statement may be added to the statement being built.
+     * If the statement breaks any rules, this will throw an error.
+     */
+    private check(statement: AtomicStatementV2) {
+        if (this.schema) {
+            verifyAtomicStatement(statement, this.statements, this.schema);
+        }
+    }
+
+    /**
+     * Add to the statement, that the given attribute should be in the given range, i.e. that lower <= attribute < upper.
+     * @param attribute the attribute that should be checked
+     * @param lower: the lower end of the range, inclusive.
+     * @param upper: the upper end of the range, exclusive.
+     * @returns the updated builder
+     */
+    addRange(attribute: string, lower: string, upper: string): this {
+        const statement: AtomicStatementV2 = {
+            type: StatementTypes.AttributeInRange,
+            attributeTag: attribute,
+            lower,
+            upper,
+        };
+        this.check(statement);
+        this.statements.push(statement);
+        return this;
+    }
+
+    /**
+     * Add to the statement, that the given attribute should be one of the values in the given set.
+     * @param attribute the attribute that should be checked
+     * @param set: the set of values that the attribute must be included in.
+     * @returns the updated builder
+     */
+    addMembership(attribute: string, set: string[]): this {
+        const statement: AtomicStatementV2 = {
+            type: StatementTypes.AttributeInSet,
+            attributeTag: attribute,
+            set,
+        };
+        this.check(statement);
+        this.statements.push(statement);
+        return this;
+    }
+
+    /**
+     * Add to the statement, that the given attribute should be one of the values in the given set.
```

Comment:
The docs here is the same as for `addMembership`. It should be the opposite of it.

Reply:
I added a \_not\_ :thinking: